backend/httpclient/http_client_test.go

```diff
@@ -87,6 +87,14 @@ func TestNewClient(t *testing.T) {
 			require.Equal(t, "mw1", usedMiddlewares[2].(MiddlewareName).MiddlewareName())
 		})
 
+		t.Run("New client should verify that middlewares are not duplicated", func(t *testing.T) {
```

Comment:
Can you please add a test that verifies the same works when adding a duplicate Middleware via opts.ConfigureMiddleware func? I don't think it does and maybe the validation should happen in roundTripperFromMiddlewares?

Reply:
a thousand places where you can configure middlewares :upside_down_face: good catch!